### PR TITLE
Use a temporary table to verify if deletes are supported

### PR DESCRIPTION
**Pull request by hashhar**

### CODE REVIEW

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduDistributedQueries.java [hunk 1]
```diff
@@ -121,6 +121,30 @@ public void testDelete()
         // TODO Support these test once kudu connector can create tables with default partitions
     }
 
+    @Override
+    public void testDeleteWithComplexPredicate()
+    {
+        // TODO Support these test once kudu connector can create tables with default partitions
```
Comment (kokosing) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduDistributedQueries.java:127]: Maybe for Kudu we could use previous assertion?

**Review by kokosing — APPROVED** [review]

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduDistributedQueries.java (hunk 1 above)
Comment (hashhar) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduDistributedQueries.java:127]: i.e. keep the override but add an assertion to catch this once Kudu starts supporting CREATE with default partitions?

```java
@Override
public void testDeleteWithComplexPredicate()
    if (!hasBehavior(SUPPORTS_DELETE)) {
        assertQueryFails("DELETE FROM nation", "This connector does not support deletes");
        return;
    }
}
```

Did I understand correctly?

Comment (kokosing) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduDistributedQueries.java:127]: Yes

**Comment by hashhar** [comment]
AC as fixup @kokosing PTAL again.

**Review by kokosing — APPROVED** [review]
Nice!

**Comment by hashhar** [comment]
auto-squashed the fixup.

**Comment by hashhar** [comment]
CI hit https://github.com/trinodb/trino/issues/8477.

**Pull request merged**